src/client.ts

```diff
@@ -467,13 +475,19 @@ export class Client {
 
   private incPublisherId() {
     const publisherId = this.publisherId
-    this.publisherId++
+    if (++this.publisherId >= MAX_PUBLISHERS_OR_SUBSCRIBERS) {
```

Comment:
I'm not sure this would be the best approach, since this is an always increasing index and the publishers (as the consumers) can also be destroyed. We keep, on the client, a map of active consumers/publishers for other purposes: we could use the size of the map in order to determine whether we open a new connection or not. Or maybe you already considered that and found some issue with this approach that I'm failing to spot right now, let me know,